esm/deserialize.js

```diff
@@ -17,11 +19,15 @@ const deserializer = ($, _) => {
     if ($.has(index))
       return $.get(index);
 
-    const [type, value] = _[index];
+    const [type, value, arg2, arg3] = _[index];
```

Comment:
this bails out array optimizations as there are always 2 entries and only in one case 4, not the other way around.

Dare I say in this case I would prefer all extra info/details stored as value instead of having a variable length but, if that's somehow not ideal or problematic, please use `...args` instead so that syntax doesn't try to access indexes that don't exist, thanks.

Reply:
Hmm, switching to `...args` is easy

> I would prefer all extra info/details stored as value instead of having a variable length but

will see if i am able to follow what you mean...
currently when i do: 
```js
console.log(serialize(new Uint8Array([3,2,1])))
// then i get this
[ [ 'Uint8Array', 2, 0, 3 ], [ 0, 'AwIB==' ], [ 9, 1 ] ]
```
but you want me to store it as something like this?
```js
[ [ 'Uint8Array', [ [ 9, 1 ], 0, 3 ] ], [ 0, 'AwIB==' ] ]
```
so that i can run it as this?
```js
return as(new env[type](...value), index);
```

Reply:
anyway (have switched to `...args` for now)

Reply:
you got the hint right ... if it's not too much trouble, see if it works too! The typed array name grants it should not affect other cases